packages/app-store/_utils/useAddAppMutation.ts

```diff
@@ -40,16 +40,18 @@ function useAddAppMutation(_type: App["type"] | null, options?: Parameters<typeo
 
     const json = await res.json();
 
+    const externalUrl = !json.url.startsWith(window.location.origin);
+
     if (!isOmniInstall) {
       window.location.href = json.url;
-      return;
+      return { showToast: !externalUrl && !json.url.endsWith("/setup") };
```

Comment:
So, we are assuming here that if /setup redirection is happening, then there is something more required to do before successful installation.
This is true for Sendgrid but not for other apps like Zapier. In Zapier's case, the app is installed immediately and the user needs to configure it inside Zapier.

But I like the idea of being able to do it automatically without explicit configuration by each app. 👍👍

Also, I think it would be better to rename `showToast` to something more generic because we can choose the characteristic to do more things.
e.g. `showToast` -> Tells that you can show the toast
If we rename it to something like `setupPending`, we can use it show a message on the redirected page as well that 'Your installation is pending till this setup is complete.'